### PR TITLE
fixing issue #23

### LINKED ISSUE #23: make API more "static" like
https://github.com/kenzanlabs/keystone/issues/7

### PR DESCRIPTION
**Pull request by cxoltero**
kenzanlabs/keystone#7
fixes #23

### CODE REVIEW

**Comment by obuckley-kenzan** [comment]
Can one of the admins verify this patch?

File: src/index.js [hunk 1]
```diff
@@ -32,38 +32,36 @@ var config = {
...
-
-  var pipeline = {
-    test: nodeTest()
-  };
+    return testPipeline();
+  }
+};
 
-  return pipeline;
+function testPipeline() {
```
Comment (obuckley-kenzan) [inline src/index.js]: should we apply our `pipelineFactory` convention here too?

File: src/index.js [hunk 2]
```diff
@@ -32,38 +32,36 @@ var config = {
...
+    .pipe(istanbul.writeReports, config.plugins.istanbul)
+    .pipe(istanbul.enforceThresholds, config.plugins.istanbul);
 
-  function nodeTest() {
-    handyman.log('Running mocha tests');
+  return stream();
+}
 
-    nodeCoverage();
+function nodeCoverage() {
```
Comment (obuckley-kenzan) [inline src/index.js]: I would prefer we add a verb here, eg. `generateCoverageReport` or something along those lines

**Comment by obuckley-kenzan** [comment]
This looks nice @cxoltero !

**Comment by obuckley-kenzan** [comment]
ok to test

**Comment by obuckley-kenzan** [comment]
@cxoltero 
Shouldn't the README be updated as well?

File: src/index.js [hunk 3]
```diff
@@ -32,38 +32,36 @@ var config = {
...
-    nodeCoverage();
+function generateNodeCoverageReport() {
 
-    return lazypipe()
-      .pipe(mocha, config.plugins.mocha)
-      .pipe(istanbul.writeReports, config.plugins.istanbul)
-      .pipe(istanbul.enforceThresholds, config.plugins.istanbul);
-  }
+  return gulp.src(config.files.src)
+    // Covering files
```
Comment (paul-barry-kenzan) [inline src/index.js]: Please remove the comments here. Devs will be able to understand what each pipe is doing based on the code.

**Pull request merged**